### ISSUE TITLE
Implement permutation weighting

### ISSUE THREAD

**Issue by CamDavidsonPilon** [pzivich/zEpid#54]
I think it would fit well in `ipw/`, and the method looks quite easy to implement. 

https://arxiv.org/pdf/1901.01230.pdf

**Comment by pzivich** [comment]
Yeah focusing on balancing the covariates seems like a good strategy for the IPTW approach. The paper you linked looks like a good start. I will have to read more about the continuous treatments in that paper (since I am still not clear how continuous treatments operate in a IPW framework)

One of the students at UNC talked about an R package he has been working on dealing with balance for propensity scores. It does some interesting things like setting the maximum allowable unbalance to exist. You can vary this threshold to see if you have any gains in efficiency relative to allowing more unbalance
https://github.com/ngreifer/optweight
It uses a quadratic solver for the optimization problem. The interesting item to me is the longitudinal optimal weights, since that is a slightly harder problem. Since this is more complex (and a little outside my experience since I haven't used quadratic solvers), the paper you linked is probably the better start
